### DISCUSSION TITLE
Official vim plugin

### DISCUSSION THREAD

**Discussion by taphill** [architect/architect#1415] [Feature Requests]
I wasn't sure the best place to ask this, but is there any interest in having an "official" vim plugin for adding syntax highlighting to `.arc` files? There are links on the README for vscode, sublime, and atom plugins under the Architect organization , but nothing from vim/neovim. I think it would be nice to have.

My company just recently started using Architect so I created a plugin for this since I couldn't find one. I'd be happy to contribute it if it's something people want!

**Comment by ryanblock** [comment]
Yes, please! This would be super rad. Any chance we can take advantage of the common Arc regex?

**Comment by taphill** [comment]
Ok awesome! I'm sure that would be possible. Vim regex seem pretty powerful, but I am new to working with them so I'm still getting used to some of the quirks. Is there an example somewhere of the regex pattern you're referring to?

Just for reference, here's what it's looking like so far on my machine. Not sure if this is close to what you'd be hoping for? Open to any feedback!

<img width="533" alt="Screenshot 2023-04-03 at 5 12 04 PM" src="https://user-images.githubusercontent.com/62676321/229638507-2fca996d-9754-4ceb-b7a8-b31675b26871.png">

**Comment by ryanblock** [comment]
Ah, check out https://github.com/architect/syntaxes

This is where we manage all the mega-regex that powers the other syntax highlighting plugins we maintain. It's got a few very minor issues, feel free to improve it!

**Comment by taphill** [comment]
Ok cool, thank you! I will start looking into that.

**Comment by taphill** [comment]
Ok, I have updated the plugin as closely as possible based on the regex's from the repo that you linked me to. I couldn't copy them directly since the way vim implements regex is slightly different. That said, I used the `example.arc` file and I feel like what I currently have is extremely close to the other editor plugins.

Would someone be able to install this plugin and let me know if it is acceptable, or if there is anything else that needs to be tweaked?

https://github.com/taphill/arc.vim

**Reply by supaspoida** [reply]
Thanks for doing this, I'm excited to spruce up my arc files! Here is what it looks like on my machine with the Nord color theme:

<img width="760" alt="Screenshot 2023-04-07 at 8 43 06 PM" src="https://user-images.githubusercontent.com/4137/230696092-861bc283-c5da-4190-94bd-94b5356dad11.png">

**Reply by jlipps** [reply]
![Screen Shot 2023-04-08 at 17 07 56](https://user-images.githubusercontent.com/605053/230747644-a2f5b946-bb54-4e6a-a5dd-e1fae788dc60.png)

Here's how mine looks. I can't tell if it's intended or if I'm even doing arc right (but it has seemed to work). Looks like a bit of discrepancy in highlighting for e.g. `@aws` vs `@plugins`. not sure why. At any rate this is a big improvement over nothing!

**Reply by colepeters** [reply]
Looking good for me, thanks so much! 
<img width="373" alt="image" src="https://user-images.githubusercontent.com/1713932/230931430-c4843acb-fe48-41bf-828a-0ed69e56cb25.png">

**Reply by taphill** [reply]
That's odd @jlipps. Is it possible that some sort of extra blank Unicode character is being added after those pragmas that aren't being highlighted? I'm not sure what would be causing that, but that shouldn't be happening

**Comment by taphill** [comment]
Thanks for the initial feedback everyone! I would love for this plugin to live within the Architect organization so that others can easily find it/contribute to it. Is that something that would be possible and/or desired?

**Comment by ryanblock** [comment]
Definitely, can we coordinate in the Architect discord? https://discord.com/invite/y5A2eTsCRX

**Reply by taphill** [reply]
Sure!

**Reply by ryanblock** [reply]
→ https://github.com/architect/vim-plugin
